src/atlases.jl

```diff
@@ -0,0 +1,123 @@
+
+"""
+    abstract type AbstractAtlas end
+
+An abstract class for atlases.
+"""
+abstract type AbstractAtlas end
+
+"""
+    struct RetractionAtlas{
+        TInvRetr<:AbstractInverseRetractionMethod,
+        TRetr<:AbstractRetractionMethod,
+        TBasis<:AbstractBasis,
+    } <: AbstractAtlas
+
+An atlas indexed by points on a manifold, such that coordinate transformations are performed
+using retractions, inverse retractions and coordinate calculation for a given basis.
+"""
+struct RetractionAtlas{
+    TInvRetr<:AbstractInverseRetractionMethod,
+    TRetr<:AbstractRetractionMethod,
+    TBasis<:AbstractBasis,
+} <: AbstractAtlas
+    invretr::TInvRetr
+    retr::TRetr
+    basis::TBasis
+end
+
+function RetractionAtlas(
```

Comment:
We could provide and document a default, that uses `exp/log`?

Reply:
Yes, that's a good idea.